src/list/list-heading.js

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright (c) Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+
+import React from 'react';
+import ReactIs from 'react-is';
+
+import {getOverrides} from '../helpers/overrides.js';
+
+import {
+  StyledHeadingRoot,
+  StyledHeadingContent,
+  StyledHeadingContentRow,
+  StyledHeadingEndEnhancerContainer,
+  StyledHeadingEndEnhancerDescriptionContainer,
+  StyledHeadingMainHeading,
+  StyledHeadingSubHeading,
+} from './styled-components.js';
+import type {HeadingPropsT} from './types.js';
+
+function RenderNode(props) {
```

Comment:
👍 This node | abstractcomponent approach sgtm. Maybe pull this out into a shared utility in a follow up PR